### ISSUE TITLE
integration.states.test_file.FileTest.test_directory_broken_symlink - cannot serialize 'set' object

### ISSUE THREAD

**Issue by waynew** [saltstack/salt#52912]
2018.3 failed [salt-centos-7-py2-tcp](https://jenkinsci.saltstack.com/job/2018.3/job/salt-centos-7-py2-tcp/22/testReport/junit/integration.states.test_file/FileTest/test_directory_broken_symlink), [salt-centos-7-py3-tcp](https://jenkinsci.saltstack.com/job/2018.3/job/salt-centos-7-py3-tcp/17/testReport/junit/integration.states.test_file/FileTest/test_directory_broken_symlink), [salt-ubuntu-1604-py2-tcp](https://jenkinsci.saltstack.com/job/2018.3/job/salt-ubuntu-1604-py2-tcp/17/testReport/junit/integration.states.test_file/FileTest/test_directory_broken_symlink)
---

can not serialize 'set' object

```

Traceback (most recent call last):
  File "/tmp/kitchen/testing/tests/integration/states/test_file.py", line 1210, in test_directory_broken_symlink
    file_mode=644, dir_mode=755)
  File "/tmp/kitchen/testing/tests/support/case.py", line 843, in run_state
    ret = self.run_function('state.single', [function], **kwargs)
  File "/tmp/kitchen/testing/tests/support/case.py", line 817, in run_function
    kwarg=kwargs)
  File "/tmp/kitchen/testing/salt/client/__init__.py", line 754, in cmd
    **kwargs)
  File "/tmp/kitchen/testing/salt/client/__init__.py", line 358, in run_job
    raise SaltClientError(general_exception)
SaltClientError: can not serialize 'set' object
		
```

**Comment by s0undt3ch** [comment]
@dmurphy18 I have a fix for this one, _stealing_ the ticket.

**Comment by s0undt3ch** [comment]
PR's for 2018.3 and 2019.2.1 submitted and linked to this issue

**Comment by s0undt3ch** [comment]
A link to the 2019.2.1 failure https://jenkinsci.saltstack.com/job/2019.2.1/view/Transport/job/salt-centos-7-py2-tcp/4/testReport/junit/integration.states.test_file/FileTest/test_directory_broken_symlink

**Comment by s0undt3ch** [comment]
This is fixed in 2019.2.1, for 2018.3 https://github.com/saltstack/salt/pull/53349 still needs to be merged
